tests/acceptance/tests/Settings/BasicConsentBannerAcceptAllCookies.spec.ts

```diff
@@ -0,0 +1,25 @@
+import { test } from '@fixtures/AcceptanceTest';
+
+test('As a shop customer, I want to use a accept all cookies button in the basic cookie consent banner in storefront.', { tag: '@Settings' }, async ({
+    ShopCustomer,
+    StorefrontHome,
+    TestDataService,
+    DefaultSalesChannel,
+}) => {
+
+    await TestDataService.createSystemConfigEntry('core.basicInformation.acceptAllCookies', true, DefaultSalesChannel.salesChannel.id);
+
+    await ShopCustomer.goesTo(StorefrontHome.url());
+    await ShopCustomer.expects(StorefrontHome.consentCookieBannerContainer).toBeVisible();
+    await ShopCustomer.expects(StorefrontHome.consentAcceptAllCookiesButton).toBeVisible();
+    await StorefrontHome.consentAcceptAllCookiesButton.click();
+
+    let allCookies = await StorefrontHome.page.context().cookies();
+    ShopCustomer.expects(allCookies.length).toEqual(2);
+
+    await StorefrontHome.page.reload();
+
+    allCookies = await StorefrontHome.page.context().cookies();
+    ShopCustomer.expects(allCookies.length).toEqual(4);
+    await ShopCustomer.expects(StorefrontHome.consentCookieBannerContainer).not.toBeVisible();
```

Comment:
To enhance the readability and structure of test cases, adding test.step can be a great approach. It helps to break down the test into logical, understandable steps, making it easier for others to read and maintain. It'd also be great that, people will understand what the test case is doing in detail without understanding the code.

```suggestion
    await test.step('Set system configuration to accept all cookies', async () => {
    await TestDataService.createSystemConfigEntry('core.basicInformation.acceptAllCookies', true, DefaultSalesChannel.salesChannel.id);
});

await test.step('Navigate to the homepage and verify cookie consent banner', async () => {
    await ShopCustomer.goesTo(StorefrontHome.url());
    await ShopCustomer.expects(StorefrontHome.consentCookieBannerContainer).toBeVisible();
    await ShopCustomer.expects(StorefrontHome.consentAcceptAllCookiesButton).toBeVisible();
});

await test.step('Accept all cookies', async () => {
    await StorefrontHome.consentAcceptAllCookiesButton.click();
    let allCookies = await StorefrontHome.page.context().cookies();
    ShopCustomer.expects(allCookies.length).toEqual(2);  // Verify initial cookies
});

await test.step('Reload page and verify cookies', async () => {
    await StorefrontHome.page.reload();
    let allCookies = await StorefrontHome.page.context().cookies();
    ShopCustomer.expects(allCookies.length).toEqual(4);  // Verify cookies after reload
});

await test.step('Verify cookie consent banner is no longer visible', async () => {
    await ShopCustomer.expects(StorefrontHome.consentCookieBannerContainer).not.toBeVisible();
});
```